packages/loader/container-loader/src/container.ts

```diff
@@ -753,6 +758,40 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     }
 
     public async attach(request: IRequest): Promise<void> {
+        let success = false;
+        let numRetries = 0;
+        let retryAfter = 1;
+        let id: string | undefined;
+        do {
+            try {
+                await this.attachCore(request);
```

Comment:
i don't think this is safe. we should be more specific about what we retry. probably only need to retry to the network call, and i think we usually do that in the driver

Reply:
like i don;t think we can fire the attaching event multiple time, and we def can;t fire attaching after attached which is possible with the generalized retry here

Reply:
But there is only 1 network call in the attach flow and if that succeeds then we don't expect anything else to fail. So if attach network call succeeds and anything else fails, then I don't think retrying would do much.

Reply:
Added here as it is central place for attach rather than in all drivers.

Reply:
We already take care of events firing. Like we only fire attaching if the container is detached in attach core.
I think I will add check to fire attached only from attaching.
If already attached, then we don't execute attach core as we just return.

Reply:
just retry the network call then. also, needs tests regardless

Reply:
Added tests and retried only the createContainer call in attach. Actually deduped some of the common code in the package.